src/pyupgrade/plugins/convert_typed_dict_functional_to_class.rs

```diff
@@ -23,10 +24,18 @@ fn match_typed_dict_assign<'a>(
                 keywords,
             } = &value.node
             {
-                if let ExprKind::Name { id: func_name, .. } = &func.node {
-                    if func_name == "TypedDict" {
-                        return Some((class_name, args, keywords));
+                match &func.node {
+                    ExprKind::Name { id: func_name, .. } => {
+                        if func_name == "TypedDict" {
+                            return Some((class_name, args, keywords, &func.node));
+                        }
                     }
+                    ExprKind::Attribute { attr, .. } => {
+                        if attr == "TypedDict" {
+                            return Some((class_name, args, keywords, &func.node));
+                        }
+                    }
+                    _ => {}
                 }
```

Comment:
We have `crate::ast::helpers::match_module_member` for doing this more correctly.

Reply:
Nice one, thanks!